qark/plugins/js_interface_plugin.py

```diff
@@ -0,0 +1,95 @@
+import sys
+import os
+import re
+
+sys.path.insert(0, os.path.dirname(os.path.realpath(__file__)) + '../lib')
+
+from yapsy.IPlugin import IPlugin
+from plugins import PluginUtil
+from modules import common
+from lib.pubsub import pub
+
+
+class JsInterfacePlugin(IPlugin):
+    # regex to determine if file contains WebView
+    webViewRegex = r'android.webkit.WebView'
+
+    # regexes to get WebView names
+    argRegex = r'(android\.webkit\.)?WebView\s(\w*?)[,)]'
+    declRegex = r'(android\.webkit\.)?WebView\s(\w*?)[;|\s]=?'
+
+    # regexes to match addJavascriptInterface() calls
+    inlineRegex = r'new\s(android\.webkit\.)?WebView\(.*?\)\.addJavascriptInterface\(.*?\)'
+
+    # returns a list of WebView argument variable names
+    def getWebViewArgNames(self, fileBody):
+        argNames = []
+        for match in re.finditer(self.argRegex, fileBody):
+            argNames.append(match.group(2))
+        return argNames
+
+    # returns a list of WebView declared variable names
+    def getWebViewDeclarationNames(self, fileBody):
+        declNames = []
+        for match in re.finditer(self.declRegex, fileBody):
+            declNames.append(match.group(2))
+        return declNames
+
+    # True if fileBody contains an inline addJavascriptInterface() call i.e. new WebView(..).addJavascriptInterface(),
+    # False otherwise
+    def containsInlineJavascriptInterfaceCall(self, fileBody):
+        return re.search(self.inlineRegex, fileBody) is not None
+
+    # True if file contains an addJavascriptInterface() call on the given WebView, False otherwise
+    def containsJavascriptInterfaceCall(self, fileBody, webViewVarName):
+        # escape w
+        if webViewVarName == 'w':
+            webViewVarName = r'\\w'
+        regex = r'%s\.addJavascriptInterface\(.*?\)' % webViewVarName
+        return re.search(regex, fileBody) is not None
+
+    def createIssueDetails(self, fileName):
+        return 'addJavascriptInterface() called on instance of WebView in file: %s\n' \
```

Comment:
I'd like @tushardalvi to weigh in on the wording of the Issue. I'm not entirely sure on
```'Expose addJavascriptInterface() only to web pages from which all input is trustworthy.' ```